### ISSUE TITLE
Broken MdDialog with angular beta 16 and ng2-material 0.3.7

### ISSUE THREAD

**Issue by Maks3w** [justindujardin/ng2-material#204]
It works with angular beta 15 but not with beta 16

```
angular2.dev.js:24831 EXCEPTION: Error: Uncaught (in promise): EXCEPTION: Error in :0:0
ORIGINAL EXCEPTION: TypeError: Cannot read property 'length' of undefined
ORIGINAL STACKTRACE:
TypeError: Cannot read property 'length' of undefined
    at _flattenStyles (http://localhost:3000/node_modules/angular2/bundles/angular2.dev.js:8950:31)
    at new DomRenderer (http://localhost:3000/node_modules/angular2/bundles/angular2.dev.js:8740:22)
    at DomRootRenderer_.DomRootRenderer.renderComponent (http://localhost:3000/node_modules/angular2/bundles/angular2.dev.js:8719:20)
    at DebugDomRootRenderer.renderComponent (http://localhost:3000/node_modules/angular2/bundles/angular2.dev.js:4630:50)
    at new MdDialog (http://localhost:3000/node_modules/ng2-material/components/dialog/dialog.js:32:39)
    at AppView.Object.defineProperty.get (AppComponent_Host.template.js:24:63)
    at AppView._View_AppComponent_Host0.injectorGetInternal (AppComponent_Host.template.js:42:77)
    at AppView.injectorGet (http://localhost:3000/node_modules/angular2/bundles/angular2.dev.js:22677:23)
    at ElementInjector.get (http://localhost:3000/node_modules/angular2/bundles/angular2.dev.js:15728:29)
    at ElementInjector.get (http://localhost:3000/node_modules/angular2/bundles/angular2.dev.js:15731:44)
ERROR CONTEXT:
[object Object]
```

```ts
// app.component.ts
import {MdDialog} from "ng2-material/components/dialog/dialog";
@Component(
    {
        selector: 'documents-app',
        providers: [MdDialog],
        // ...
    }
)

```

```ts
// error-dialog.component.ts
import {ElementRef, Injectable} from "angular2/core";
import {MdDialog, MdDialogConfig, MdDialogBasic, MdDialogRef} from "ng2-material/components/dialog/dialog";

@Injectable()
export class ErrorDialogComponent {
    private dialog: MdDialog;

    constructor(dialog: MdDialog) {
        this.dialog = dialog;
    }

    showErrorMessage(element: ElementRef, message: string, actionText?: string, actionMethod?: () => void): void {
        let config = new MdDialogConfig()
            .textContent(message)
            .clickOutsideToClose(false)
            .title('Error')
            .ok(actionText)
            .cancel('Close');

        this.dialog.open(MdDialogBasic, element, config)
            .then(
                (ref: MdDialogRef) => {
                    ref.whenClosed.then(
                        (result) => {
                            if (result) {
                                actionMethod();
                            }
                        }
                    )
                }
            );
    };
}
```

**Comment by axal90** [comment]
`E:/node/msls/node_modules/ng2-material/components/dialog/dialog.d.ts(1,60): error TS2305: Module '"E:/node/msls/node_modules/angular2/core"' has no exported member 'ResolvedProvider'.` 
Seems like they changed something: [Angular 2 Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
A fix should be comming soon

**Comment by railsstudent** [comment]
Fall back to angular2-beta-14 and project runs fine.

**Comment by justindujardin** [comment]
I took a look at this issue, and unfortunately there are some bugs in beta 17 that stop me from supporting it. They made changes to the `DynamicComponentLoader` which is partly workable, but they also changed the way content projection works for `ng-content` elements, which breaks use-cases that use ng-content and other directives at the same time (https://github.com/angular/angular/issues/8344)

I will lock the build to a compatible version for now, and once Beta 18 is released I will continue to upgrade to it.

**Comment by justindujardin** [comment]
Thanks for bringing this up.  I'm going to close this issue and track work for its fix under #207
